src/clients/python/src/tigerbeetle_python/client.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+import asyncio
+import ctypes
+import logging
+import os
+import random
+import threading
+import time
+from collections.abc import Callable  # noqa: TCH003
+from dataclasses import dataclass
+from typing import Any
+
+from . import bindings
+
+logger = logging.getLogger("tigerbeetle")
+
+
+class AtomicInteger:
+    def __init__(self, value=0):
+        self._value = value
+        self._lock = threading.Lock()
+
+    def increment(self):
+        with self._lock:
+            self._value += 1
+            return self._value
+
+
+@dataclass
+class CompletionContextSync:
+    event: threading.Event
+
+
+@dataclass
+class CompletionContextAsync:
+    loop: asyncio.AbstractEventLoop
+    event: asyncio.Event
+
+
+@dataclass
+class InflightPacket:
+    packet: bindings.CPacket
+    response: Any
+    operation: bindings.Operation
+    c_event_type: Any
+    c_result_type: Any
+    on_completion: Callable | None
+    on_completion_context: CompletionContextSync | CompletionContextAsync | None
+
+
+
+def tbid() -> int:
+    """
+    Generates a Universally Unique and Sortable Identifier as a 128-bit integer. Based on ULIDs.
+    """
+    time_ms = time.time_ns() // (1000 * 1000)
+    randomness = os.urandom(10)
+
+    return int.from_bytes(
+        time_ms.to_bytes(6, "big") + randomness,
+        "big",
+    )
+
+
+amount_max = (2 ** 128) - 1
+
+
+class InitError(Exception):
+    pass
+
+class PacketError(Exception):
+    pass
+
+
+class Client:
+    _clients: dict[int, Any] = {}
+    _counter = AtomicInteger()
+
+    def __init__(self, cluster_id: int, replica_addresses: str):
+        self._client_id = Client._counter.increment()
```

Comment:
whoa, shouldn't this be a random number?

Reply:
or is it a different kind of client id?

Reply:
Ahhh, it _is_ different, perhaps call it `client_key` then?

Reply:
Heh, I was extra confused here for a sec, because initially this was a random number. But yes, it is a key for looking up the client (a similar pattern is used for packets too) internally.